### ISSUE TITLE
Not able to create a module with lazy loading.

### ISSUE THREAD

**Issue by abhishekdgeek** [angular/angular-cli#9287]
### Versions
```
    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
@angular/cli: 1.4.9
node: 6.11.2
os: darwin x64
@angular/animations: 4.4.6
@angular/common: 4.4.6
@angular/compiler: 4.4.6
@angular/core: 4.4.6
@angular/forms: 4.4.6
@angular/http: 4.4.6
@angular/platform-browser: 4.4.6
@angular/platform-browser-dynamic: 4.4.6
@angular/router: 4.4.6
@angular/cli: 1.4.9
@angular/compiler-cli: 4.4.6
@angular/language-service: 4.4.6
typescript: 2.3.4
```

### Repro steps

* Create a new app - `ng new customer-app --routing`
* Try to create a new module via CLI - `ng generate module customers --routing`

### Observed behavior
```
TypeError: Cannot read property 'NullLogger' of undefined
    at SchematicEngine.createContext (/Applications/XAMPP/xamppfiles/htdocs/customer-app/node_modules/@angular-devkit/schematics/src/engine/engine.js:64:38)
    at SchematicImpl.call (/Applications/XAMPP/xamppfiles/htdocs/customer-app/node_modules/@angular-devkit/schematics/src/engine/schematic.js:35:38)
    at Promise (/Applications/XAMPP/xamppfiles/htdocs/customer-app/node_modules/@angular/cli/tasks/schematic-run.js:72:23)
    at Class.run (/Applications/XAMPP/xamppfiles/htdocs/customer-app/node_modules/@angular/cli/tasks/schematic-run.js:71:16)
    at Class.run (/Applications/XAMPP/xamppfiles/htdocs/customer-app/node_modules/@angular/cli/commands/generate.js:149:33)
    at resolve (/Applications/XAMPP/xamppfiles/htdocs/customer-app/node_modules/@angular/cli/ember-cli/lib/models/command.js:261:20)
    at Class.validateAndRun (/Applications/XAMPP/xamppfiles/htdocs/customer-app/node_modules/@angular/cli/ember-cli/lib/models/command.js:240:12)
    at Promise.resolve.then.then (/Applications/XAMPP/xamppfiles/htdocs/customer-app/node_modules/@angular/cli/ember-cli/lib/cli/cli.js:140:24)
```

### Desired behavior
Create a module with routes of itself within it.

Is this only in Angular 1.5+?

**Comment by shekharramola** [comment]
me too not able to create

**Comment by abhishekdgeek** [comment]
@sravaniperumalla that will get you latest cli, I want to work in Angular version 4 not version 5.

**Comment by yogZ-codeR** [comment]
+1

**Comment by abhishekdgeek** [comment]
Seems like it's not just me - https://github.com/angular/angular-cli/issues/9303.

**Comment by yogZ-codeR** [comment]
@abhishekdgeek - the CLI update worked for me and it just updated the CLI version and not the other core angular modules.

Still not sure why it stopped to work in previous version all of a sudden.

**Comment by abhishekdgeek** [comment]
Yes! @yogZ-codeR my problem was resolved the very same day with 1.4.10 I am just curious how this is happening as I have used 1.4.9 for all my projects!

**Comment by hansl** [comment]
Fixed by a release of CLI 1.4.
